tests/subsys/bluetooth/mesh/scheduler_model/message_validity/src/main.c

```diff
@@ -0,0 +1,571 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdint.h>
+#include <stdlib.h>
+#include <zephyr/ztest.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/timeutil.h>
+#include <bluetooth/mesh/gen_onoff_srv.h>
+#include <bluetooth/mesh/time_srv.h>
+#include <bluetooth/mesh/scheduler_srv.h>
+#include <bluetooth/mesh/scheduler_cli.h>
+#include <scheduler_internal.h>
+#include <model_utils.h>
+#include <time_util.h>
+#include <sched_test.h>
+
+#define TIMEOUT 5
+
+static uint8_t test_idx;
+static int msg_sent_cnt;
+static struct tm start_tm;
+struct bt_mesh_schedule_entry valid_entry;
+
+struct bt_mesh_msg_ctx test_ctx = {
+	.addr = 1,
+};
+
+static struct bt_mesh_time_srv time_srv = BT_MESH_TIME_SRV_INIT(NULL);
+
+static void onoff_set(struct bt_mesh_onoff_srv *srv, struct bt_mesh_msg_ctx *ctx,
+		      const struct bt_mesh_onoff_set *set, struct bt_mesh_onoff_status *rsp)
+{
+}
+
+static void onoff_get(struct bt_mesh_onoff_srv *srv, struct bt_mesh_msg_ctx *ctx,
+		      struct bt_mesh_onoff_status *rsp)
+{
+}
+
+struct bt_mesh_onoff_srv_handlers onoff_srv_handlers = {
+	.set = onoff_set,
+	.get = onoff_get,
+};
+
+struct bt_mesh_onoff_srv onoff_srv = BT_MESH_ONOFF_SRV_INIT(&onoff_srv_handlers);
+
+static void srv_action_set_cb(struct bt_mesh_scheduler_srv *srv, struct bt_mesh_msg_ctx *ctx,
+			      uint8_t idx, struct bt_mesh_schedule_entry *entry)
+{
+	ztest_check_expected_value(srv->model);
```

Comment:
Why do you check `srv->model` and not `srv` itself? I feel like as long as a pointer to the correct srv is passed, checking that this srv hasn't gotten its model pointer changed is out of scope for this test? Same for `cli`.